src/NewChaiBuffer.hpp

```diff
@@ -286,23 +333,23 @@ class NewChaiBuffer : public chai::CHAICopyable
 
 private:
 
-  template< typename U=T >
-  std::enable_if_t< !std::is_base_of< CHAICopyable, U >::value >
-  moveInnerData( chai::ExecutionSpace const LVARRAY_UNUSED_ARG( space ) )
+  template< typename U=T_non_const >
+  std::enable_if_t< !internal::HasMemberFunction_move< U > >
+  moveInnerData( chai::ExecutionSpace const, std::ptrdiff_t const, bool const )
   {}
 
-  template< typename U=T >
```

Comment:
@rrsettgast This is a bug that I believe is in develop. There are two issues here first is that in the for-loop below you loop up to the capacity of the buffer, which means you might be accessing uninitialized objects if the size is less than the capacity. The second is that if you have `array1d< SortedArray< T > >` and you call move on it then you'll be calling the `SortedArray` copy constructor which does a deep copy.